suite/tests/runcmp.cmake

```diff
@@ -51,38 +52,37 @@ if (cmd_result)
   message(FATAL_ERROR "*** ${cmd} failed (${cmd_result}): ${cmd_err}***\n")
 endif (cmd_result)
 
-# get expected output
-# we assume it has already been processed w/ regex => literal, etc.
-file(READ "${cmp}" str)
+set(output "${cmd_out}")
+if ("${capture}" STREQUAL "stderr")
+  set(output "${cmd_err}")
+endif ()
+
+set(tmp "${cmp}-out")
+file(WRITE "${tmp}" "${output}")
 
-# We do not support regex b/c ctest can't handle big regex:
+# We do not support regex in expect file b/c ctest can't handle big regex:
 #   "RegularExpression::compile(): Expression too big."
-# so we use STREQUAL.
 
 # We used to implement a "diff -b" via cmake regex on the output and expect
 # file, but that gets really slow (30s for 750KB strings) so now we require
-# strict matches.
+# strict matches per line.
 
-set(output "${cmd_out}")
-if ("${capture}" STREQUAL "stderr")
-  set(output "${cmd_err}")
-endif ()
+# Use diff -I to get the ability to skip some lines.
 
-if (NOT "${output}" STREQUAL "${str}")
-  # make it easier to debug
-  set(tmp "${cmp}-out")
-  file(WRITE "${tmp}" "${output}")
-  set(tmp2 "${cmp}-expect")
-  file(WRITE "${tmp2}" "${str}")
-
-  set(diffcmd "diff")
-  execute_process(COMMAND ${diffcmd} ${tmp} ${tmp2}
+set(diffcmd "diff")
```

Comment:
This is problematic for Windows where there may not be a "diff" command; plus for UNIX it may not be on the path.  Previously this was just a hacky piece of code only ever run on failure.  But to run "diff" every time it should be done properly: use `find_program` to find it (to try to locate if not on the path); it still could cause problems for Windows developers.  Even if the GA CI VM's have a "diff" installed, is that standard on Windows?  I doubt it.  It might be best to either do a line-by-line compare and implement the ignore-line manually in CMake code (this fits better with the general philosophy: see below), or use the original code for Windows.

Note that generally the philosophy has been to rely on as few external applications as possible, primarily motived by Windows where often nothing but CMake is installed, but also Mac where utilities don't have GNU extensions.  Generally we have preferred doing everything in CMake, and not relying on little utility applications being installed.

Reply:
Thanks for the detailed review, I get the idea. Let's reimplement it fully in CMake then. I'm not familiar with CMake, so it will take some time.

Reply:
There are examples of walking line by line in our other code: e.g.:
```
$ git grep 'foreach.*line' 
core/CMake_symbol_check.cmake:foreach(line ${lines})
```

Reply:
> Let's reimplement it fully in CMake then.

Done.